test/testlib/nuodb_database_utilities.go

```diff
@@ -311,3 +315,37 @@ func CheckRestoreRequests(t *testing.T, namespaceName string, podName string, da
 	}
 	return restoreRequest, legacyRestoreRequest
 }
+
+func CreateNginxDeployment(t *testing.T, namespaceName string) {
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespaceName)
+	k8s.RunKubectl(t, kubectlOptions, "create", "deployment", NGINX_DEPLOYMENT, "--image=nginx")
+	k8s.RunKubectl(t, kubectlOptions, "create", "service", "clusterip", NGINX_DEPLOYMENT, "--tcp=80:80")
+	podName := GetPodName(t, namespaceName, NGINX_DEPLOYMENT)
+	AwaitPodUp(t, namespaceName, podName, 60*time.Second)
+
+	AddDiagnosticTeardown(TEARDOWN_DATABASE, t, func() {
```

Comment:
this should not be reusing the DATABASE teardown.
It should also provide its own teardown which deletes the deployment and the clusterIP service. Otherwise, it leaves garbage in the namespace.

Reply:
Good point!